pkgs/cronet_http/lib/cronet_client.dart

```diff
@@ -34,10 +122,27 @@ import 'src/messages.dart';
 /// }
 /// ```
 class CronetClient extends BaseClient {
-  static late final HttpApi _api = HttpApi();
+  CronetEngine? _engine;
+  final bool _ownedEngine;
+
+  CronetClient([CronetEngine? engine])
+      : _engine = engine,
+        _ownedEngine = engine == null;
+
+  @override
+  void close() {
+    if (_ownedEngine) {
```

Comment:
would `_engine != null` be sufficient here making _ownedEngine property unnecessary?

Reply:
`_engine` gets assigned in `send` if not set in the constructor.